### PR TITLE
refactor: [M3-6802] - Remove Linodes Redux store

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
## Description 📝
Completes the RQ Linodes migration by removing the Linodes Redux store and refactoring the last few places that still used it.

## Major Changes 🔄
- Deleted Linode Redux store (`src/store/linodes`)
- Migrated `BackupDrawer` and `SelectPlanPanel` to RQ Linode hooks/HOCs
- Moved Linodes utilities to new `src/utilities/linodes.ts` file
- Slightly refactored `ConfigSelect` to display no options and error messages

## How to test 🧪
1. Ensure all actions on Linodes (i.e., create, rename, reboot, delete, migrate, etc.) continue to work as expected 
2. Ensure enabling/disabling backups (for single Linode, for all Linodes, auto-enroll) all continue to work as expected
3. Ensure `SelectPlanPanel` continues to work as expected
4. Ensure all other components in the App that fetch Linodes continue to work as expected, e.g.:
    - Setting up NodeBalancers for Linodes
    - Attaching volumes to Linodes
    - Adding Linodes to Firewalls
    - Creating Kubernetes clusters
5. Ensure unit tests and E2E test pass

### CODE REVIEW

File: packages/manager/src/components/AuthenticationWrapper/AuthenticationWrapper.tsx [hunk 1]
```diff
@@ -128,9 +125,6 @@ export class AuthenticationWrapper extends React.Component<CombinedProps> {
         queryFn: getAccountSettings,
         queryKey: 'account-settings',
       }),
-
-      // Fetch first page of Linodes (TODO: remove once the Linodes RQ migration is complete)
-      this.props.getLinodesPage(),
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/AuthenticationWrapper/AuthenticationWrapper.tsx:133]: Linodes are now fetched at the component level with RQ hooks.

File: packages/manager/src/components/OrderBy.tsx [hunk 2]
```diff
@@ -19,6 +19,8 @@ import {
 export interface OrderByProps<T> extends State {
   data: T[];
   handleOrderChange: (orderBy: string, order: Order) => void;
+  order: Order;
+  orderBy: string;
```
Comment (hkhalil-akamai) [inline packages/manager/src/components/OrderBy.tsx:23]: These were being provided anyways, I'm not sure why they weren't in the interface.

File: packages/manager/src/features/Linodes/LinodesCreate/LinodeCreateContainer.tsx [hunk 3]
```diff
@@ -768,11 +771,6 @@ class LinodeCreateContainer extends React.PureComponent<CombinedProps, State> {
           });
         }
 
-        /** if cloning a Linode, upsert Linode in redux */
-        if (createType === 'fromLinode') {
-          this.props.upsertLinode(response);
-        }
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/LinodeCreateContainer.tsx:774]: This is handled on the RQ side through the Linodes event handler.

File: packages/manager/src/features/Linodes/LinodesCreate/deriveDefaultLabel.test.ts
Comment (hkhalil-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/deriveDefaultLabel.test.ts:1]: Refactored this from a HOC to a utility function.

File: packages/manager/src/features/Volumes/VolumeDrawer/ConfigSelect.tsx [hunk 4]
```diff
@@ -33,70 +31,61 @@ const ConfigSelect: React.FC<CombinedProps> = (props) => {
...
-    if (configList.length === 1) {
+    if (configList?.length === 1) {
       const newValue = configList[0].value;
       if (value !== newValue) {
         onChange(configList[0].value);
       }
     }
   }, [configList, onChange, value]);
 
-  if (configs.length < 1) {
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Volumes/VolumeDrawer/ConfigSelect.tsx:73]: This slightly changes the behavior of `ConfigSelect`: before, it wouldn't render if there were no configs available (or if there was an error fetching configs); now, it renders as long as a Linode ID is provided, displaying a message if there are no available options or an error message if an error occurs while retrieving configs.

File: packages/manager/src/features/Volumes/VolumeDrawer/ConfigSelect.tsx [hunk 5]
```diff
@@ -33,70 +31,61 @@ const ConfigSelect: React.FC<CombinedProps> = (props) => {
...
+          error ?? configsError
+            ? 'An error occurred while retrieving configs for this Linode.'
+            : undefined
+        }
+        noOptionsMessage={
+          () =>
+            !configs || configs.length == 0
+              ? 'No configs are available for this Linode.'
+              : 'No options.' // No matches for search
+        }
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/Volumes/VolumeDrawer/ConfigSelect.tsx:72]: This is the only non-indentation change in this block.

File: packages/manager/src/queries/linodes/linodes.ts [hunk 6]
```diff
@@ -102,14 +100,6 @@ export const useLinodeUpdateMutation = (id: number) => {
...
 };
 
-export const useAllLinodeConfigsQuery = (id: number, enabled = true) => {
-  return useQuery<Config[], APIError[]>(
-    [queryKey, 'linode', id, 'configs'],
-    () => getAllLinodeConfigs(id),
-    { enabled }
-  );
-};
-
```
Comment (hkhalil-akamai) [inline packages/manager/src/queries/linodes/linodes.ts:112]: Moved to `configs.ts`.

File: packages/manager/src/queries/linodes/linodes.ts [hunk 7]
```diff
@@ -218,15 +203,6 @@ export const useLinodeChangePasswordMutation = (id: number) =>
...
   );
 
-export const useLinodeDeleteMutation = (id: number) => {
-  const queryClient = useQueryClient();
-  return useMutation<{}, APIError[]>(() => deleteLinode(id), {
-    onSuccess() {
-      queryClient.invalidateQueries([queryKey]);
-    },
-  });
-};
```
Comment (hkhalil-akamai) [inline packages/manager/src/queries/linodes/linodes.ts:228]: Exact duplicate of `useDeleteLinodeMutation`.

**Review by bnussman-akamai — APPROVED** [review]
Very nice. Feels so good to finally be at this point. Let's make sure e2es all pass before we merge

**Review by jaalah-akamai — APPROVED** [review]

**Pull request merged**